src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/personalrelationships/profiledetails/ContactPersonProfileDetailsSyncIntTest.kt

```diff
@@ -0,0 +1,244 @@
+import com.github.tomakehurst.wiremock.client.WireMock.putRequestedFor
+import com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo
+import com.github.tomakehurst.wiremock.matching.StringValuePattern
+import kotlinx.coroutines.test.runTest
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.check
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.isNull
+import org.mockito.kotlin.verify
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR
+import org.springframework.http.HttpStatus.NOT_FOUND
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.integration.SqsIntegrationTestBase
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.BookingProfileDetailsResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.PrisonerProfileDetailsResponse
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomissync.model.ProfileDetailsResponse
+
+class ContactPersonProfileDetailsSyncIntTest(
+  @Autowired private val nomisApi: ContactPersonProfileDetailsNomisApiMockServer,
+  @Autowired private val dpsApi: ContactPersonProfileDetailsDpsApiMockServer,
+  @Autowired private val syncService: ContactPersonProfileDetailsSyncService,
+) : SqsIntegrationTestBase() {
+
+  @Nested
+  @DisplayName("OFFENDER_PHYSICAL_DETAILS-CHANGED for contact person details")
+  inner class DomesticStatusChanged {
+    @Nested
+    inner class HappyPath {
+      @Test
+      fun `should sync new profile details`() = runTest {
+        nomisApi.stubGetProfileDetails(
+          "A1234AA",
+          nomisResponse(
+            offenderNo = "A1234AA",
+            bookings = listOf(
+              booking(
+                bookingId = 12345,
+                startDateTime = "2024-09-03T12:34:56",
+                endDateTime = null,
+                latestBooking = true,
+                profileDetails = listOf(
+                  profileDetails(
+                    type = "MARITAL",
+                    code = "M",
+                    createDateTime = "2024-09-04T12:34:56",
+                    createdBy = "A_USER",
+                    modifiedDateTime = null,
+                    modifiedBy = null,
+                    auditModuleName = "NOMIS",
+                  ),
+                ),
+              ),
+            ),
+          ),
+        )
+        dpsApi.stubSyncDomesticStatus(response = dpsResponse())
+
+        // TODO replace this with a send event when listener implemented
+        syncService.profileDetailsChanged(profileType = "MARITAL", offenderNo = "A1234AA", bookingId = 12345)
```

Comment:
The plan is to get these simple test cases working with the full message flow, then expand on the test cases once the tests are e2e.